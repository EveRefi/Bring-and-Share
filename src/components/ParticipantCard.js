import React from "react";
import '../styles/ParticipantCard.css'

const ParticipantCard = ({id, name, email, toBring, dislikes, dietInfo, removeParticipant}) => {
    return(
        <>
            <div className="eachParticipant">
                <div data-testid="name" className="participantName">
                    {name}
                </div>
                <div data-testid="email" className="participantEmail">
                    <a href={`mailto:${email}`}>{email}</a>
                </div>
                <div data-testid="toBring" className="participantBrings">
                    Brings: {toBring}
                </div>
                <div data-testid="dislikes" className="participantDislikes">
                    Does not like: {dislikes}
                </div >
                <div data-testid="dietInfo" className="participantDietInfo">
                    Any other Diet information: {dietInfo}
                </div>
                <button data-testid="removeButton" className="participantDelete" onClick={()=>removeParticipant(id)}>Delete</button>
            </div>

        </>
    )
}

export default ParticipantCard