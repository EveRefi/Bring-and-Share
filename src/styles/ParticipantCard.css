@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");


.eachParticipant{
    margin: 10px;
    width: 180px;
    border: solid black;
}
/* participantName
participantToBring
participantDislikes
participantDietInfo
participantDelete */