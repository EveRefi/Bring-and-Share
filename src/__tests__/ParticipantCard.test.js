import React from "react";
import { render, getByTestId } from "@testing-library/react";
import ParticipantCard from "../components/ParticipantCard";

describe('ParticipantCard component', () => {
    const mockDate = {
        id: 1234,
        name: 'MyName',
        toBring: 'IAmBringing',
        dislikes: 'MyDislikes',
        dietInfo: 'MyDietInfo'
    } 
    const mockRemove = jest.fn()

    it('matches the snapshot', () => {
        const {asFragment} = render (
            <ParticipantCard
                id={mockDate.id}
                name={mockDate.name}
                toBring={mockDate.toBring}
                dislikes={mockDate.dislikes}
                dietInfo={mockDate.dietInfo}
                removeParticipant={mockRemove} />)

        expect(asFragment()).toMatchSnapshot()
    })

    it('gets all elements', () => {
        const {getByTestId} = render (
            <ParticipantCard
                id={mockDate.id}
                name={mockDate.name}
                toBring={mockDate.toBring}
                dislikes={mockDate.dislikes}
                dietInfo={mockDate.dietInfo}
                removeParticipant={mockRemove} />)

        expect(getByTestId('name')).toHaveClass('participantName')
        expect(getByTestId('toBring')).toHaveClass('participantBrings')
        expect(getByTestId('dislikes')).toHaveClass('participantDislikes')
        expect(getByTestId('dietInfo')).toHaveClass('participantDietInfo')
        expect(getByTestId('removeButton')).toBeInTheDocument()

    })



})