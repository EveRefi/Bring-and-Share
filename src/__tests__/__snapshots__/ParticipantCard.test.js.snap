// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantCard component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="eachParticipant"
  >
    <div
      class="participantName"
      data-testid="name"
    >
      MyName
    </div>
    <div
      class="participantEmailInfo"
      data-testid="email"
    >
      <a
        href="mailto:undefined"
      />
    </div>
    <div
      class="participantBrings"
      data-testid="toBring"
    >
      Brings: IAmBringing
    </div>
    <div
      class="participantDislikes"
      data-testid="dislikes"
    >
      Does not like: MyDislikes
    </div>
    <div
      class="participantDietInfo"
      data-testid="dietInfo"
    >
      Any other Diet information: MyDietInfo
    </div>
    <button
      class="participantDelete"
      data-testid="removeButton"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;
